### PR TITLE
Loki: Clean up Language provider and remove unused code

### PR DESCRIPTION
**Pull request by ivanahuckova**
Language provider had a lot of code that wasn't used anymore as we moved from `slate` => `monaco`. This PR cleans it up and removes everything that is not used anymore.  Plus it adds comments and `todos` on what to focus on next.

### CODE REVIEW

File: public/app/plugins/datasource/loki/syntax.ts [hunk 1]
```diff
@@ -191,8 +191,8 @@ export const BUILT_IN_FUNCTIONS = [
 ];
 
 export const FUNCTIONS = [...AGGREGATION_OPERATORS, ...RANGE_VEC_FUNCTIONS, ...BUILT_IN_FUNCTIONS];
-export const LOKI_KEYWORDS = [...FUNCTIONS, ...PIPE_OPERATORS, ...PIPE_PARSERS].map((keyword) => keyword.label);
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/syntax.ts:194]: I checked and this was not used anymore anywhere

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 2]
```diff
@@ -1,77 +1,25 @@
...
 
-const DEFAULT_KEYS = ['job', 'namespace'];
-const EMPTY_SELECTOR = '{}';
-const HISTORY_ITEM_COUNT = 10;
-const HISTORY_COUNT_CUTOFF = 1000 * 60 * 60 * 24; // 24h
 const NS_IN_MS = 1000000;
 
-// When changing RATE_RANGES, check if Prometheus/PromQL ranges should be changed too
-// @see public/app/plugins/datasource/prometheus/promql.ts
-const RATE_RANGES: CompletionItem[] = [
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:31]: CompletionItem is used in slate editor that we don't use anymore, as we replaced it with Monaco. 
image

I have double checked if autocompletion + completion in query builder still works as expected. 

image

image

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 3]
```diff
@@ -1,77 +1,25 @@
...
-  { label: '$__auto', sortValue: '$__auto' },
-  { label: '1m', sortValue: '00:01:00' },
-  { label: '5m', sortValue: '00:05:00' },
-  { label: '10m', sortValue: '00:10:00' },
-  { label: '30m', sortValue: '00:30:00' },
-  { label: '1h', sortValue: '01:00:00' },
-  { label: '1d', sortValue: '24:00:00' },
-];
-
-export const LABEL_REFRESH_INTERVAL = 1000 * 30; // 30sec
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:41]: This wasn't used anywhere.

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 4]
```diff
@@ -1,77 +1,25 @@
...
-    const lastQueried = dateTime(recent.ts).fromNow();
-    hint = `${hint} Last queried ${lastQueried}.`;
-  }
-
-  return {
-    ...item,
-    documentation: hint,
-  };
-}
-
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:68]: All of this code was for slate editor to suggest previously run queries in autocomplete, but we don't use it anymore. All of the autocompletion logic is now in `CompletionDataProvider`, so it is safe to remove. https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts#L32

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 5]
```diff
@@ -91,13 +39,6 @@ export default class LokiLanguageProvider extends LanguageProvider {
     Object.assign(this, initialValues);
   }
 
-  // Strip syntax chars
-  cleanText = (s: string) => s.replace(/[{}[\]="(),!~+\-*/^%\|]/g, '').trim();
-
-  getSyntax(): Grammar {
-    return syntax;
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:98]: Another methods previously used in slate that are not used anywhere.

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 6]
```diff
@@ -127,210 +67,6 @@ export default class LokiLanguageProvider extends LanguageProvider {
     return this.labelKeys;
   }
 
-  /**
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:130]: This logic was fully replaced in https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts#L32

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 7]
```diff
@@ -1,77 +1,25 @@
...
-    documentation: hint,
-  };
-}
-
 export default class LokiLanguageProvider extends LanguageProvider {
   labelKeys: string[];
   labelFetchTs: number;
   started = false;
   datasource: LokiDatasource;
-  lookupsDisabled = false; // Dynamically set to true for big/slow instances
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:74]: `lookupsDisabled` is setting used in Prometheus for `low data Prometheus mode`, but in Loki we are not setting it up anywhere, so it is always just `false`and therefore can be removed.

File: public/app/plugins/datasource/loki/LanguageProvider.ts [hunk 8]
```diff
@@ -1,77 +1,24 @@
...
-
-  return {
-    ...item,
-    documentation: hint,
-  };
-}
-
 export default class LokiLanguageProvider extends LanguageProvider {
   labelKeys: string[];
-  labelFetchTs: number;
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LanguageProvider.ts:71]: This wasn't used anywhere. Left over of a functionality where we were deciding where to refresh labels based on `labelFetchTs`.

**Review by matyax — APPROVED** [review]
Fantastic! 👏 👏

**Pull request merged**